### ISSUE TITLE
Wonolog support?

### ISSUE THREAD

**Issue by strarsis** [xwp/stream#1040]
Is support for [Wonolog](https://github.com/inpsyde/Wonolog) planned?
This makes it much easier for plugins to log in a unified way.

**Comment by kasparsd** [comment]
Thanks for opening the issue @strarsis! I hadn't seen Wonolog before -- it looks like a great tool.

I some way it is very similar to Stream in providing the logging functionality but it doesn't include the UI for the records and requires developers to setup the log points.

I really like the concept of `context` in Wonolog which supports an array of data instead of just the strings as in Stream.

What do you think would be the ideal integration for Wonolog? In theory, we could have a dedicated plugin that hooks into the Stream logger and just passes the data to Wonolog directly. Looks like we would need to add a hook [in the `log()` method](https://github.com/xwp/stream/blob/f3c2396c7bedfc7a3f0f6bf3340fff3fa74afad0/classes/class-log.php#L62-L152) that other plugins could use to override or adjust the record store.

**Comment by strarsis** [comment]
@kasparsd: Yes, IMHO `wonolog` should be used as an internal, unified logging mechanism for core and plugins to be used, while plugins like `Stream` can additionally hook into `wonolog` and provide a UI, log storage mechanism and other features like email alerts. The advantage is that new plugins just can use `wonolog` and don't need to handle different audit/logging plugins and versions and the audit/logging plugins automatically support all new plugins.

**Comment by lkraav** [comment]
Whatever the platform choice, I fully support the idea of "we need a unified logging platform".

**Comment by strarsis** [comment]
A [notification API is proposed for WordPress core](https://make.wordpress.org/core/2016/10/03/feature-project-proposal-notifications-api/), the same would be nice for logging.

**Comment by lkraav** [comment]
Ha, and looks like I've advocated for Stream right there, back in 2016 https://make.wordpress.org/core/2016/10/03/feature-project-proposal-notifications-api/#comment-31197

But https://github.com/xwp/stream/issues/762 died.

**Comment by strarsis** [comment]
@lkraav: Maybe it will work this time using `monolog`. 😸
